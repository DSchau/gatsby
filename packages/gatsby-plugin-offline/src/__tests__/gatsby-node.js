const { onPreBootstrap } = require(`../gatsby-node`)

const getOptions = config => {
  return {
    reporter: {
      stripIndent: jest.fn(str => str.trim()),
    },
    store: {
      getState() {
        return {
          config,
        }
      },
    },
  }
}

test(`it throws error if assetPrefix is in config`, () => {
  expect(() =>
    onPreBootstrap(
      getOptions({
        assetPrefix: `https://cdn.example.com`,
      })
    )
  ).toThrowErrorMatchingSnapshot()
})

test(`it does not throw if assetPrefix is not in config`, () => {
  expect(() =>
    onPreBootstrap(
      getOptions({
        pathPrefix: `/blog`,
      })
    )
  ).not.toThrow()
})
