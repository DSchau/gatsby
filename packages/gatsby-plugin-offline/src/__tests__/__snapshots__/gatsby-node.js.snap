// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it throws error if assetPrefix is in config 1`] = `
"gatsby-plugin-offline cannot be used with the assetPrefix option.
      A service worker specifies that it must be served on the same domain.
      Either remove the assetPrefix from gatsby-config.js or remove this plugin."
`;
